pkg/apis/enterprisesearch/v1beta1/enterprisesearch_types.go

```diff
@@ -98,20 +98,38 @@ func (ent *EnterpriseSearch) SetAssociationConf(assocConf *commonv1.AssociationC
 	ent.assocConf = assocConf
 }
 
-func (ent *EnterpriseSearch) AssociationStatus() commonv1.AssociationStatus {
-	return ent.Status.Association
+func (ent *EnterpriseSearch) RequiresAssociation() bool {
+	return ent.Spec.ElasticsearchRef.Name != ""
 }
 
-func (ent *EnterpriseSearch) SetAssociationStatus(status commonv1.AssociationStatus) {
-	ent.Status.Association = status
+func (ent *EnterpriseSearch) GetAssociations() []commonv1.Association {
+	associations := make([]commonv1.Association, 0)
+	if ent.Spec.ElasticsearchRef.IsDefined() {
+		associations = append(associations, ent)
+	}
+	return associations
 }
 
-func (ent *EnterpriseSearch) RequiresAssociation() bool {
-	return ent.Spec.ElasticsearchRef.Name != ""
+func (ent *EnterpriseSearch) ID() int {
+	return 0
```

Comment:
I think the same argument can be made for other funcs too, so I do what we did elsewhere - rename it to indicate it's Associations ID. We might do what you suggest for ent and kb, but if we want to do that I think it could be a separate change to reduce the scope here, WDYT?